rust/core-lib/src/tabs.rs

```diff
@@ -371,9 +372,13 @@ impl<W: Write + Send + 'static> Documents<W> {
             let buffer_id = self.next_buffer_id();
             self.new_empty_view(rpc_peer, &view_id, buffer_id);
         }
+
+        // schedule plugin notification for after the view has received an identifer
+        let view_id2 = view_id.to_owned();
         let init_info = self.buffers.lock().editor_for_view(&view_id)
             .unwrap().plugin_init_info();
-        self.plugins.document_new(&view_id, init_info);
+        self.plugins.run_on_idle(move |p| p.document_new(&view_id2, &init_info));
```

Comment:
We need the mutex because each plugin runloop runs in a separate thread.

I'm not sure about enforcing sequential ordering on notifications; I could imagine situations where we explicitly _do_ want to send out notifications while holding on to a response, for instance if we want to do some `will_save` notification. 

I think `run_next_iteration` on `RpcPeer` could be a reasonable solution, though.

Also I was slightly confused: I just went digging through some code and there's actually _already_ an `available_plugins` request that the client sends to core when it gets its initial view identifier; and core doesn't send any notifications as part of the `document_new`; instead that function launches plugins, which when they connect attempt to notify the client that they've launched.

So at the very least that existing `available_plugins` request could be an `init_view` request, although we could also remove it and consolidate behind some new solution that offers correct message ordering.

Anyway I've got to run, but I'll pick this up tomorrow.